src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+            feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+          }
+        });
+        setUnFollow(data);
+      } else {
+        const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: false};
+        feedsContext.feeds[indexFeed] = newFeed;
+        unfollowUserAnon(dataFollowAnon);
+      }
+    } else if (!feedData?.anon_user_info_color_name) {
+      feedsContext?.feeds.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: true};
+        }
+      });
+      setFollow(data);
+    } else {
+      const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: true};
+      feedsContext.feeds[indexFeed] = newFeed;
+      followUserAnon(dataFollowAnon);
+    }
+    setMainFeeds(feedsContext.feeds, feedsContextDispatch);
+  };
+
+  const followUnfollowTopic = async (item: any, topic: string, offset: number) => {
+    const id = removePrefixTopic(topic);
+    const feedTopic = feedsContext.topicFeeds
+      ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(id))
+      : [];
+    const feedMain = feedsContext.feeds;
+
+    const user_id = item?.actor?.id;
+    const username = item?.actor?.data?.username;
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: user_id,
+      username_follower: myProfile?.username,
+      username_followed: username,
+      follow_source: 'feed'
+    };
+    const dataFollowAnon = {
+      follow_source: 'post',
+      post_id: item?.id
+    };
+    const indexFeed = feedTopic?.findIndex((feed) => {
+      return feed?.id === item?.id;
+    });
+    const feedData = feedTopic[indexFeed];
+    if (feedData?.is_following_target) {
+      if (!feedData?.anon_user_info_color_name) {
+        feedTopic.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedTopic[index] = {...feedTopic[index], is_following_target: false};
+          }
+        });
+        feedMain.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedMain[index] = {...feedMain[index], is_following_target: false};
+          }
+        });
+        setUnFollow(data);
+      } else {
+        feedTopic[indexFeed] = {...feedTopic[indexFeed], is_following_target: false};
+        feedMain[indexFeed] = {...feedMain[indexFeed], is_following_target: false};
+        unfollowUserAnon(dataFollowAnon);
+      }
+    } else if (!feedData?.anon_user_info_color_name) {
+      feedTopic.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedTopic[index] = {...feedTopic[index], is_following_target: true};
+        }
+      });
+      feedMain.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedMain[index] = {...feedMain[index], is_following_target: true};
+        }
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You are directly mutating the state (`feedsContext.feeds`, `feedTopic`, `feedMain`) which is not recommended in React. Instead, you should create a new copy of the state with the updated values and then set this new state. 

```diff
- feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+ const newFeeds = [...feedsContext.feeds];
+ newFeeds[index] = {...newFeeds[index], is_following_target: false};
+ // Then set this new state using your state update function

- feedTopic[index] = {...feedTopic[index], is_following_target: false};
+ const newFeedTopic = [...feedTopic];
+ newFeedTopic[index] = {...newFeedTopic[index], is_following_target: false};
+ // Then set this new state using your state update function

- feedMain[index] = {...feedMain[index], is_following_target: false};
+ const newFeedMain = [...feedMain];
+ newFeedMain[index] = {...newFeedMain[index], is_following_target: false};
+ // Then set this new state using your state update function
```

56, 120:
You are setting the state (`feedsContext.feeds`, `feedMain`) directly with the mutated array. Instead, you should use the dispatch function (`feedsContextDispatch`) to update the state.

```diff
- setMainFeeds(feedsContext.feeds, feedsContextDispatch);
+ feedsContextDispatch(setMainFeeds(newFeeds));

- setMainFeeds(feedMain, feedsContextDispatch);
+ feedsContextDispatch(setMainFeeds(newFeedMain));
```

38, 42, 50, 54, 95, 99, 112, 116:
The async functions `setUnFollow`, `unfollowUserAnon`, `setFollow`, and `followUserAnon` are called without await or handling their promise. If these functions are asynchronous and return a promise, it would be better to handle them properly to catch any potential errors and avoid unhandled promise rejections.

```diff
- setUnFollow(data);
+ await setUnFollow(data);

- unfollowUserAnon(dataFollowAnon);
+ await unfollowUserAnon(dataFollowAnon);

- setFollow(data);
+ await setFollow(data);

- followUserAnon(dataFollowAnon);
+ await followUserAnon(dataFollowAnon);
```

Overall, the `followUnfollow` and `followUnfollowTopic` functions are quite large and complex. Consider breaking them down into smaller, more manageable functions. This will make your code easier to read, test, and maintain.


<!-- This is an auto-generated comment by OSS CodeRabbit -->